cdap-operational-stats-core/src/test/java/co/cask/cdap/operations/cdap/CDAPOperationalStatsTest.java

```diff
@@ -86,21 +88,25 @@ public void test() throws Exception {
     Assert.assertEquals(4, entities.getDatasets());
     Assert.assertEquals(1, entities.getStreams());
     Assert.assertEquals(0, entities.getStreamViews());
-    CDAPTransactions transactions = new CDAPTransactions();
+    final CDAPTransactions transactions = new CDAPTransactions();
     transactions.initialize(injector);
     Assert.assertEquals(AbstractCDAPStats.SERVICE_NAME, transactions.getServiceName());
     Assert.assertEquals("transactions", transactions.getStatType());
-    // wait for 5 secs to make sure the metrics get gauged
-    TimeUnit.SECONDS.sleep(5);
-    transactions.collect();
+    // wait for some time to make sure the metrics get gauged
+    Tasks.waitFor(true, new Callable<Boolean>() {
+      @Override
+      public Boolean call() throws Exception {
+        transactions.collect();
+        return transactions.getNumCommittingChangeSets() > 0;
```

Comment:
better to have other metrics in this condition too, just to be absolutely sure? ``return transactions.getNumCommittingChangeSets() > 0 && transactions.getNumInProgressTransactions() > 0 && transactions.getNumInvalidTransactions() > 0``